sql/src/main/java/io/crate/operation/scalar/arithmetic/TrigonometricFunctions.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.operation.scalar.arithmetic;
+
+import com.google.common.collect.Iterables;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionInfo;
+import io.crate.operation.scalar.DoubleScalar;
+import io.crate.operation.scalar.ScalarFunctionModule;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+
+import java.util.Collections;
+import java.util.function.DoubleUnaryOperator;
+
+public final class TrigonometricFunctions {
+
+    private static final Iterable<DataType> SUPPORTED_INPUT_TYPES =
+        Iterables.concat(DataTypes.NUMERIC_PRIMITIVE_TYPES, Collections.singletonList(DataTypes.UNDEFINED));
+
+    public static void register(ScalarFunctionModule module) {
+        register(module, "sin", Math::sin);
+        register(module, "asin", ((DoubleUnaryOperator) Math::asin).compose(TrigonometricFunctions::checkRange));
```

Comment:
`register(module, "asin", x -> Math.asin(checkRange(x)));` would probably be easier to read?